tensorboard/webapp/metrics/views/card_renderer/scalar_card_container.ts

```diff
@@ -283,21 +300,28 @@ export class ScalarCardContainer implements CardRenderer, OnInit {
         this.store.select(getMetricsScalarSmoothing),
         this.store.select(getMetricsXAxisType)
       ),
-      switchMap(([runsData, smoothing, xType]) => {
+      switchMap(([runsData, smoothing, xScaleType]) => {
         const dataSeriesList = runsData.map(({runId, points}) => {
-          return {
-            id: runId,
-            points: points.map((point) => {
-              return {
+          points = points.map((point) => {
+            // Convert wallTime in seconds to milliseconds.
+            // TODO(stephanwlee): when the legacy line chart is removed, do the conversion at
+            // the effects.
+            const wallTime = point.wallTime * 1000;
+            const x = xScaleType === XAxisType.STEP ? point.x : wallTime;
+            return {...point, x, wallTime};
+          });
+          switch (xScaleType) {
```

Comment:
An if statement would be clearer:

if (xScaleType === XAxisType.RELATIVE and points.length) {
  // do stuff.
}